File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let myMap = new Map();
    for(let i=0; i<nums.length; i++){
        let val =  target-nums[i];
        if(myMap.has(val))
        return [myMap.get(val),i];
     
            myMap.set(nums[i], i);
        
    }
    return [];
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        res = []
        for i in range(len(nums)):
            val = dict.get(target-nums[i])
            if val != None:
                res.append(i)
                res.append(val)
            else:
                dict[nums[i]] = i
        return res            